arrow-core-data/src/main/kotlin/arrow/typeclasses/FunctorFilter.kt

```diff
@@ -16,19 +14,25 @@ interface FunctorFilter<F> : Functor<F> {
   /**
    * A combined map and filter. Filtering is handled via [Option] instead of [Boolean] such that the output type [B] can be different than the input type [A].
    */
+  @Deprecated("Please use mapNotNull")
   fun <A, B> Kind<F, A>.filterMap(f: (A) -> Option<B>): Kind<F, B>
```

Comment:
Should we give this a default impl of ` = mapNotNull { a -> Option.fromNullable(f(a)) }` for the time of deprecation?

Reply:
You just gave me an idea. To extract `Option` to a separate module, I only need to remove it from public API. Internal usages can be replaced (temporary of course) with `Either<Unit, A>`, unblocking the `Option` extraction.